code/modules/power/tesla/energy_ball.dm

```diff
@@ -11,98 +23,132 @@
 	. = ..()
 	energy = starting_energy
+	miniball = is_miniball
 	START_PROCESSING(SSobj, src)
-	var/turf/spawned_turf = get_turf(src)
-	message_admins("A tesla has been created at [ADMIN_VERBOSEJMP(spawned_turf)].")
-	investigate_log("(tesla) was created at [AREACOORD(spawned_turf)].", INVESTIGATE_ENGINES)
 
-/obj/anomaly/energy_ball/ex_act(severity, target)
-	return
+	if (!is_miniball)
+		set_light_on(TRUE)
+
+		var/turf/spawned_turf = get_turf(src)
+		message_admins("A tesla has been created at [ADMIN_VERBOSEJMP(spawned_turf)].")
+		investigate_log("was created at [AREACOORD(spawned_turf)].", INVESTIGATE_ENGINE)
+
+/obj/energy_ball/Destroy()
+	if(orbiting && istype(orbiting.parent, /obj/energy_ball))
+		var/obj/energy_ball/parent_energy_ball = orbiting.parent
+		parent_energy_ball.orbiting_balls -= src
 
-/obj/anomaly/energy_ball/Destroy()
 	QDEL_LIST(orbiting_balls)
 	STOP_PROCESSING(SSobj, src)
-	. = ..()
 
-/obj/anomaly/energy_ball/process(delta_time)
-	handle_energy(delta_time)
+	return ..()
 
-	move_the_basket_ball(4 + orbiting_balls.len * 1.5)
+/obj/energy_ball/ex_act(severity, target)
+	return
 
-	playsound(src.loc, 'sound/magic/lightningbolt.ogg', 100, 1, extrarange = 30)
+/obj/energy_ball/process(delta_time)
+	if(orbiting)
+		energy = 0 // ensure we dont have miniballs of miniballs
+	else
+		handle_energy(delta_time)
 
-	pixel_x = 0
-	pixel_y = 0
+		if(last_move <= world.time + 6 SECONDS)
+			last_move = world.time
+			move(4 + (orbiting_balls.len * 1.5 ))
 
-	//Main one can zap
-	//Tesla only zaps if the tick usage isn't over the limit.
-	if(!TICK_CHECK)
-		tesla_zap(src, 7, TESLA_DEFAULT_POWER, TESLA_ENERGY_PRIMARY_BALL_FLAGS)
-	else
-		//Weaker, less intensive zap
-		tesla_zap(src, 4, TESLA_DEFAULT_POWER, TESLA_ENERGY_MINI_BALL_FLAGS)
-		pixel_x = -32
-		pixel_y = -32
-		return
+			playsound(src.loc, 'sound/magic/lightningbolt.ogg', 70, TRUE, extrarange = 30)
 
-	pixel_x = -32
-	pixel_y = -32
-	for (var/ball in orbiting_balls)
-		if(TICK_CHECK)
-			return
-		var/range = rand(1, CLAMP(orbiting_balls.len, 3, 7))
-		//Miniballs don't explode.
-		tesla_zap(ball, range, TESLA_MINI_POWER/7*range, TESLA_ENERGY_MINI_BALL_FLAGS)
+			pixel_x = 0
+			pixel_y = 0
+			shocked_things.Cut(1, shocked_things.len / 1.3)
+			var/list/shocking_info = list()
 
-/obj/anomaly/energy_ball/examine(mob/user)
+			//Main one can zap
+			//Tesla only zaps if the tick usage isn't over the limit.
+			if(!TICK_CHECK)
+				tesla_zap(src, 3, TESLA_DEFAULT_POWER, shocked_targets = shocking_info)
+			else
+				tesla_zap(src, 3, TESLA_DEFAULT_POWER, ZAP_DEFAULT_FLAGS)
+				pixel_x = -32
+				pixel_y = -32
+				return
+
+			pixel_x = -32
+			pixel_y = -32
+			for (var/ball in orbiting_balls)
+				var/range = rand(1, CLAMP(orbiting_balls.len, 2, 7))
+				var/list/temp_shock = list()
+				//We zap off the main ball instead of ourselves to make things looks proper
+				tesla_zap(src, range, TESLA_MINI_POWER/7*range, shocked_targets = temp_shock)
```

Comment:
The miniball zap flags need to be passed here in order to make the miniballs not have explosion causing zaps